client-js/app/scripts/activities.ts

```diff
@@ -74,8 +74,8 @@ export class DatavaultActivity implements Activity {
 
   async start(): Promise<ActivityState> {
     console.info('Loading datavault:', this.path);
-    this.api.newDir.add(x => this.onNewDir(), this.lifetime);
-    this.api.newDataset.add(x => this.onNewDataset(), this.lifetime);
+    this.api.newDir.add(x => this.onNewDir(x), this.lifetime);
+    this.api.newDataset.add(x => this.onNewDataset(x), this.lifetime);
```

Comment:
If you look at 115 and 124 you can see that the new dir and new dataset functions now actually receive the name of the new dataset. This lets them pass just the latest item to the grapher rather than regenerating the whole list by calling the api like it used to.